### PR TITLE
feature / NUI 5068 fix label rendering in case we have many steps in the wizard

### PR DESCRIPTION
**Pull request by dmytrofirsov**
* responsive label rendering in case dynamic steps are being added/removed
* numerous edge case issues fixed
* markup for the headers in popovers has been developed

### CODE REVIEW

**Review by ivankozlovskyi — APPROVED** [review]

File: packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts [hunk 1]
```diff
@@ -41,10 +80,236 @@ export class WizardHorizontalComponent extends WizardDirective implements OnInit
...
+
+    private get isCurrentStepVisible(): boolean {
+        return this.visibleSteps.includes(this.stepsArray[this.selectedIndex]);
+    }
+
+    private get numberOfVisibleItems(): number {
+        return Math.ceil((this.headerContainerWidth - this.headerPaddings * 2) / this.stepHeaderWidth) - 1;
+    }
+
+    private handleDynamicHeaderChanges(visibleItemsNUmber: number) {
```
Comment (Vita-Karas) [inline packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts:215]: Please, add the missing return type for functions in this file

File: packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts [hunk 2]
```diff
@@ -41,10 +80,236 @@ export class WizardHorizontalComponent extends WizardDirective implements OnInit
...
+                    // This covers an edge case with the saved wizard state. It can happen that the active element
+                    // appears inside the hidden list of items, when the dynamnic steps are re-added.
+                    // We make it visible again by retrieving it back to the visible items array
+                    if (this.overflownStepsStart.length && !this.isCurrentStepVisible) {
+                        do {
+                            this.takeLastAddFirst(this.overflownStepsStart, this.visibleSteps);
+                            this.takeLastAddFirst(this.visibleSteps, this.overflownStepsEnd);
+                        }
+                        while(!this.isCurrentStepVisible)
+                    }
```
Comment (Vita-Karas) [inline packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts:194]: Can be changed to
`if (this.overflownStepsStart.length) {
          while(!this.isCurrentStepVisible){
                   this.takeLastAddFirst(this.overflownStepsStart, this.visibleSteps);
                   this.takeLastAddFirst(this.visibleSteps, this.overflownStepsEnd);
           }
}`

Comment (dmytrofirsov) [reply packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts:194]: The reason `do - while` fits there better is that if we've entered that if statement, we have to run this code at least once anyway. With the `while` going first we'll need to do the `this.isCurrentStepVisible` check twice in a row which is not ideal.

Comment (dmytrofirsov) [reply packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts:194]: on the other hand, maybe the difference not as big as I thought

File: packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts (hunk 1 above)
Comment (dmytrofirsov) [reply packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts]: added

File: packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts (hunk 2 above)
Comment (dmytrofirsov) [reply packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts:193]: changed

**Review by Vita-Karas — APPROVED** [review]

**Review by dmytrognatyshyn — APPROVED** [review]

**Review by DmytroPutrin — APPROVED** [review]

**Pull request merged**